habitat/datasets/pointnav/pointnav_dataset.py

```diff
@@ -84,19 +99,32 @@ def __init__(self, config: Optional[Config] = None) -> None:
 
         # Read separate file for each scene
         dataset_dir = os.path.dirname(datasetfile_path)
-        scenes = config.CONTENT_SCENES
-        if ALL_SCENES_MASK in scenes:
-            scenes = self.__class__._get_scenes_from_folder(
-                content_scenes_path=self.content_scenes_path,
-                dataset_dir=dataset_dir,
-            )
-
-        for scene in scenes:
-            scene_filename = self.content_scenes_path.format(
-                data_path=dataset_dir, scene=scene
+        if os.path.exists(
+            self.content_scenes_path.split("{scene}")[0].format(
+                data_path=dataset_dir
             )
-            with gzip.open(scene_filename, "rt") as f:
-                self.from_json(f.read(), scenes_dir=config.SCENES_DIR)
+        ):
+            scenes = config.CONTENT_SCENES
```

Comment:
This part of code became hard to understand. Several overloaded terms like `scenes to load` / `content_scenes` and etc. As we are adding filtering, maybe that worth to change `CONTENT_SCENES` to something else, like `SCENES_TO_LOAD` or `ALLOWED_SCENES`. Up to you, not sure how large is refactoring.

Let's add some additional variables to make the code tell a story:
```
separate_scene_files_exist =  os.path.exists(self.content_scenes_path.split("{scene}")[0].format(
                data_path=dataset_dir
            )
if separate_scene_files_exist:
...
```